wren-ai-service/eval/data_curation/utils.py

```diff
@@ -0,0 +1,426 @@
+import asyncio
+import base64
+import os
+import re
+from datetime import datetime
+from typing import Any, List, Tuple
+
+import aiohttp
+import orjson
+import sqlglot
+import sqlparse
+import streamlit as st
+import tomlkit
+from dotenv import load_dotenv
+from openai import AsyncClient
+
+load_dotenv()
+
+WREN_IBIS_ENDPOINT = os.getenv("WREN_IBIS_ENDPOINT", "http://localhost:8000")
+WREN_ENGINE_ENDPOINT = os.getenv("WREN_ENGINE_ENDPOINT", "http://localhost:8080")
+DATA_SOURCES = ["bigquery"]
+TIMEOUT_SECONDS = 60
+
+
+def get_openai_client() -> AsyncClient:
+    return AsyncClient(
+        api_key=os.getenv("OPENAI_API_KEY"),
+        timeout=TIMEOUT_SECONDS,
+    )
+
+
+def remove_limit_statement(sql: str) -> str:
+    pattern = r"\s*LIMIT\s+\d+(\s*;?\s*--.*|\s*;?\s*)$"
+    modified_sql = re.sub(pattern, "", sql, flags=re.IGNORECASE)
+
+    return modified_sql
+
+
+def add_quotes(sql: str) -> str:
+    return sqlglot.transpile(sql, read="trino", identify=True)[0]
+
+
+async def is_sql_valid(sql: str) -> Tuple[bool, str]:
+    sql = sql[:-1] if sql.endswith(";") else sql
+    async with aiohttp.request(
+        "POST",
+        f'{WREN_IBIS_ENDPOINT}/v2/connector/{st.session_state['data_source']}/query?dryRun=true',
+        json={
+            "sql": add_quotes(sql),
+            "manifestStr": base64.b64encode(
+                orjson.dumps(st.session_state["mdl_json"])
+            ).decode(),
+            "connectionInfo": st.session_state["connection_info"],
+        },
+        timeout=aiohttp.ClientTimeout(total=TIMEOUT_SECONDS),
+    ) as response:
+        if response.status == 204:
+            return True, None
+        res = await response.text()
+
+        return False, res
+
+
+async def get_validated_question_sql_pairs(
+    question_sql_pairs: list[dict],
+) -> list[dict]:
+    tasks = []
+
+    async with aiohttp.ClientSession():
+        for question_sql_pair in question_sql_pairs:
+            task = asyncio.ensure_future(is_sql_valid(question_sql_pair["sql"]))
+            tasks.append(task)
+
+        results = await asyncio.gather(*tasks)
+        return [
+            {
+                **question_sql_pairs[i],
+                "context": [],
+                "is_valid": valid,
+                "error": error,
+            }
+            for i, (valid, error) in enumerate(results)
+        ]
+
+
+def get_ddl_commands(mdl_json: dict) -> str:
```

Comment:
I'm a bit curious about how different this is from DDLConvert. If there aren't too many differences, we could consider using DDLConvert instead to reduce duplicate code and ensure consistent behavior.